### PR TITLE
Fix #4017 magnet changes do not take

### LINKED ISSUE #4017: radia: changes to some dipole parameters do not take effect
For example changing the width of the h-bend dipole pole then switching to the visualization tab does not show the change. Changing other parameters can cause things to catch up.

### LINKED ISSUE #4018: radia: allow user to subdivide extruded shapes in extrusion direction
The triangulation that Radia performs is in the plane perpendicular to the extrusion axis, leaving the shape with a single division along it. Let the user chop it up.

### PR DESCRIPTION
**Pull request by mkeilman**
Also fixes #4018, #4026

I've also tried to make inroads on #4020.  I centered the work on dipoles and extruded shapes, so there is still some cruft to support undulators, which I will change to mirror the construction of dipoles.

### CODE REVIEW

File: sirepo/template/radia.py [hunk 1]
```diff
@@ -9,11 +9,9 @@
 """
 from __future__ import division
```
Comment (e-carlin) [inline sirepo/template/radia.py:10]: This can be removed.

File: sirepo/template/radia.py [hunk 2]
```diff
@@ -358,27 +354,19 @@ def _build_clone_xform(num_copies, alt_fields, transforms):
...
+    return _update_cuboid(_build_geom_obj('cuboid', **kwargs), **kwargs)
 
 
-def _build_dipole_objects(geom, dipole, beam_axis, height_axis):
-    geom.objects.append(dipole.pole)
-    if dipole.dipoleType in ['dipoleC', 'dipoleH']:
-        geom.objects.append(dipole.magnet)
-        geom.objects.append(dipole.coil)
-        g = _update_group(dipole.corePoleGroup, [dipole.magnet, dipole.pole], do_replace=True)
+def _build_dipole_objects(geom, model, **kwargs):
```
Comment (e-carlin) [inline sirepo/template/radia.py:360]: Nice

**Review by e-carlin — COMMENTED** [review]
LGTM. One tiny nit

File: sirepo/template/radia.py (hunk 1 above)
Comment (mkeilman) [reply sirepo/template/radia.py]: 031902a

**Review by e-carlin — APPROVED** [review]

**Comment by e-carlin** [comment]
Sorry for not approving before...

**Pull request merged**